lib/ansible/galaxy/role.py

```diff
@@ -365,3 +367,24 @@ def spec(self):
         }
         """
         return dict(scm=self.scm, src=self.src, version=self.version, name=self.name)
+
+    @property
+    def requirements(self):
+        """
+        Returns role requirements
+        """
+        if self._requirements is None:
+            self._requirements = []
+            for meta_requirements in self.META_REQUIREMENTS:
+                meta_path = os.path.join(self.path, meta_requirements)
+                if os.path.isfile(meta_path):
+                    try:
+                        f = open(meta_path, 'r')
+                        self._requirements = yaml.safe_load(f)
+                    except Exception:
+                        display.vvvvv("Unable to load requirements for %s" % self.name)
+                        return False
```

Comment:
i would make this a warning and not return, the requirements would still be empty list and returned at the end, also this allows the loop to get to 2nd file and test if that works fine.

From comment above, if you process 1st file fine .. but then 2nd is error, you throw away the correct result, so combining both fixes shoudl get more reliable behaviour.

Reply:
I based myself off existing as explained here https://github.com/ansible/ansible/pull/68288#discussion_r406295476
I think that breaking on the first file found would be less confusing since you really should not have both a `yml` and `yaml` file.